tmt/steps/execute/internal.py

```diff
@@ -150,6 +170,43 @@ def check(self, test):
         else:
             return self.check_shell(test)
 
+    def _setup_reboot(self, guest):
+        """ Prepare the guest environment for potential reboot """
+        # We only want to change the environment in tmt-provisioned machines,
+        # ignore Local and Connect
+        if isinstance(guest, GuestLocal) or guest.__class__ is Guest:
+            return
+        script = self.REBOOT_SCRIPT.format(token=self.parent.reboot_token)
+        for reboot_file in self.REBOOT_SCRIPT_PATHS:
+            self.debug(f"Replacing {reboot_file} with tmt implementation")
+            guest.execute(f"echo '{script}' > {reboot_file}")
```

Comment:
This makes the debug log a bit unreadable. What about pushing the file content to the guest using `push()`?

Reply:
I can try but in order to push, it needs to be on the local system, doesn't it?

Reply:
Yes. I see two options how to do this:

* create the local file and `push()` it
* store the file in the run workdir before push and `guest.execute("cp template /usr/bin/...")`

We could also package the `rstrnt-reboot` template in a similar way as we do for `run.sh` and just replace the token part during preparation.

Reply:
Or, another option could also be to move this particular action to a deeper debug level ;-)

Reply:
I am not very keen on creating `/usr/bin/rstrnt-reboot` locally, that doesn't seem right either ;) Looking at the `guest.push()` method it is possible to set destintaion independently of the source file, so we could do something like `guest.push('template', '/usr/bin/rstrnt-reboot')` from the local workdir. That may be something worth exploring (or yeah, just use higher debug level...)

Reply:
> I am not very keen on creating `/usr/bin/rstrnt-reboot` locally

Ah, I didn't mean to package it under `/usr/bin/` but rather to do it the `run.sh` way:

```shell
> rpm -ql python3-tmt | grep run.sh
/usr/lib/python3.9/site-packages/tmt/steps/execute/run.sh
```

From here the template could be copied, adjusted and pushed to the guest.